apps/server/src/shared/repo/news/news.repo.ts

```diff
@@ -16,17 +16,33 @@ export class NewsRepo extends BaseRepo<News> {
 	/**
 	 * Find news
 	 * @param targets
-	 * @param unpublished
 	 * @param options
 	 */
-	async findAll(
+	async findAllPublished(targets: NewsTargetFilter[], options?: IFindOptions<News>): Promise<Counted<News[]>> {
+		const scope = new NewsScope();
+		scope.byTargets(targets);
+		scope.byPublished();
+
+		const countedNewsList = await this.findNewsAndCount(scope.query, options);
+		return countedNewsList;
+	}
+
+	/**
+	 * Find news
+	 * @param targets
+	 * @param creatorId - creatorId
+	 * @param options
+	 */
+	async findAllUnpublished(
 		targets: NewsTargetFilter[],
-		unpublished: boolean,
+		creatorId: EntityId,
 		options?: IFindOptions<News>
 	): Promise<Counted<News[]>> {
 		const scope = new NewsScope();
 		scope.byTargets(targets);
-		scope.byUnpublished(unpublished);
+		scope.byUnpublished();
+		scope.byCreator(creatorId);
```

Comment:
by creator might not be the correct business logic filter.
News which are not published yet, might still be visible to other teachers for example.
The business logic who has access to news which are not yet published, should be handled in the permissions check in my oppinion

Reply:
"If no creatorId is given it wouldnt be added to the query."
creator is not an optional param, so it will always want to filter by creator

Reply:
True, should be optional